### PR TITLE
Offline installation

### PR DESCRIPTION
**Pull request by dchirikov**
There are a lot of changes, but the main idea is to create TrinityX 'self-containable'
- Move from repos role to ansible module. It allows specifying the alternative location of the .rpm or .repo file.
- Add role which creates repo from plugged ISO/CD or USB-stick
- Add test scrips which install TrinityX from ISO using kickstart and set it up in an isolated environment

### CODE REVIEW

**Comment by dchirikov** [comment]
Jenkins test this please

**Comment by dchirikov** [comment]
Jenkins test this please

**Comment by dchirikov** [comment]
Jenkins test this please

**Comment by kostty** [comment]
Looks good somehow

File: site/group_vars/all [hunk 1]
```diff
@@ -133,3 +133,7 @@ enable_docker: false
 # List of additional environment modules to install
 
 additional_env_modules: []
+
+# name of the local yum repository
+#
+trinityx_local_reponame: 'trinityx-local'
```
Comment (hi-a) [inline site/group_vars/all]: Can this be renamed to `trix_local_reponame`

File: site/roles/trinity/drbd/tasks/main.yml [hunk 2]
```diff
@@ -3,9 +3,9 @@
 - name: Install DRBD packages
   yum:
     name: '{{ item }}'
-    enablerepo: elrepo
+    enablerepo: "{{ trinityx_local_reponame if local_install else 'elrepo' }}"
```
Comment (hi-a) [inline site/roles/trinity/drbd/tasks/main.yml]: I think it is best if this becomes a `when` clause

File: site/group_vars/all [hunk 3]
```diff
@@ -128,3 +132,17 @@ enable_docker: false
 # List of additional environment modules to install
 
 additional_env_modules: []
+
+# Local yum repository
+
+trinityx_local_reponame: 'trinityx-local'
+repos_port: 8080
```
Comment (hi-a) [inline site/group_vars/all]: same here

File: site/group_vars/all [hunk 4]
```diff
@@ -128,3 +132,17 @@ enable_docker: false
...
+trinityx_local_repo_baseurl: 'http://{{ trix_ctrl_ip }}:{{ repos_port }}/repos/trinityx/'
```
Comment (hi-a) [inline site/group_vars/all]: and here

File: site/roles/trinity/local_repo/handlers/main.yml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ---
-- name: reload nginx
+- name: restart nginx
```
Comment (hi-a) [inline site/roles/trinity/local_repo/handlers/main.yml]: This is actually a reload not a restart

**Comment by dchirikov** [comment]
Jenkins test this please

**Comment by dchirikov** [comment]
Jenkins test this please

**Comment by dchirikov** [comment]
Jenkins test this please

**Comment by dchirikov** [comment]
Jenkins test this please

**Pull request merged**